src/agent/Agent.ts

```diff
@@ -159,8 +162,8 @@ export class Agent {
     return this.agentConfig.mediatorUrl
   }
 
-  public async receiveMessage(inboundPackedMessage: unknown) {
-    return await this.messageReceiver.receiveMessage(inboundPackedMessage)
+  public async receiveMessage(inboundPackedMessage: unknown, transport?: Transport) {
```

Comment:
What about creating the `InboundMessageContext` here and then enhanced it with other attributes inside `MessageReciever`? 

I would rather not introduce yet another type because we already have perhaps too much (XxxMessage, XxxMessageContext, XxxPackage, ...) Then we could do the same for the outbound messages as `OutboundMessageContext`.

Reply:
Ah we already have `InboundMessageContext` of course. Yes that sounds perfect!